src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
```

Comment:
I think `Middleware.execute` should require a callback; anything that has an optional callback should take care of managing the optional-ness. Also, I don't think the `process.nextTick` is necessary here.

I would recommend changing this to:
```coffeescript
receiveMiddlewareDone = cb or ->
```